lib/mixlib/install/options.rb

```diff
@@ -21,37 +21,66 @@ class Install
     class Options
       class InvalidOptions < ArgumentError; end
 
-      attr_accessor :options
+      attr_reader :options
+      attr_reader :errors
 
-      SUPPORTED_CHANNELS = [:stable, :current]
+      OMNITRUCK_CHANNELS = [:stable, :current]
+      ARTIFACTORY_CHANNELS = [:unstable]
+      ALL_SUPPORTED_CHANNELS = OMNITRUCK_CHANNELS + ARTIFACTORY_CHANNELS
       SUPPORTED_PRODUCT_NAMES = %w[chef chefdk]
       SUPPORTED_OPTIONS = [:channel, :product_name, :product_version,
                            :platform, :platform_version, :architecture]
 
       def initialize(options)
         @options = options
+        @errors = []
         validate_options!
       end
 
       def validate_options!
-        errors = []
+        validate_product_names
+        validate_channels
+        validate_unstable_version
+        validate_platform_info
+
+        unless errors.empty?
+          raise InvalidOptions, errors.join("\n")
+        end
+      end
+
+      SUPPORTED_OPTIONS.each do |option|
+        define_method option do
+          options[option] || options[option.to_s]
+        end
+      end
+
+      private
+
+      def validate_product_names
         unless SUPPORTED_PRODUCT_NAMES.include? product_name
           errors << "Unknown product name #{product_name}. \
 Must be one of: #{SUPPORTED_PRODUCT_NAMES.join(", ")}"
         end
+      end
 
-        unless SUPPORTED_CHANNELS.include? channel
-          errors << "Unknown channel #{channel}. Must be one of: #{SUPPORTED_CHANNELS.join(", ")}"
+      def validate_channels
+        unless ALL_SUPPORTED_CHANNELS.include? channel
+          errors << "Unknown channel #{channel}. \
+Must be one of: #{ALL_SUPPORTED_CHANNELS.join(", ")}"
         end
+      end
 
-        unless errors.empty?
-          raise InvalidOptions, errors.join("\n")
+      def validate_unstable_version
+        if ARTIFACTORY_CHANNELS.include?(channel) &&
+            product_version !~ /^\d+.\d+.\d+\+[0-9]{14}$/
+          errors << "Version must match pattern '1.2.3+12345678901234' when \
+using channels #{ARTIFACTORY_CHANNELS.join(", ")}"
         end
       end
 
-      SUPPORTED_OPTIONS.each do |option|
-        define_method option do
-          options[option] || options[option.to_s]
+      def validate_platform_info
+        if platform && (!platform_version || !architecture)
```

Comment:
That was intentional.  I was trying to catch that the platform options were provided when `platform` was set, but are ignored otherwise.  If you have a strong opinion of this, I will enhance this.